browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -82,7 +82,11 @@ void UpdateContentSettingsToRendererFrames(content::WebContents* web_contents) {
             rfh->GetBrowserContext());
 
         RendererContentSettingRules rules;
-        GetRendererContentSettingRules(map, &rules);
+        content_settings::GetRendererContentSettingRules(map, &rules);
+        const GURL& primary_url =
+            rfh->GetMainFrame() ? rfh->GetMainFrame()->GetLastCommittedURL()
+                                : rfh->GetLastCommittedURL();
+        rules.FilterRulesByOutermostMainFrameURL(primary_url);
```

Comment:
shouldn't we also patch `FilterRulesByOutermostMainFrameURL` to call `FilterRulesForType` for `autoplay_rules`, `fingerprinting_rules`, `brave_shields_rules` and `cosmetic_filtering_rules`

Reply:
I meant creating a `RendererContentSettingRules::FilterRulesByOutermostMainFrameURL` calls `RendererContentSettingRules_ChromiumImpl::FilterRulesByOutermostMainFrameURL` and add brave specific types